File: Python/OneLang/Test/TestCases/OneFileTests.py
from onelang_core import *
from onelang_file import *
import OneLang.Test.TestCase as testCase

class OneFileTests:
    def __init__(self, base_dir):
        self.name = "OneFileTests"
        self.base_dir = base_dir
    
    def list_xcompiled_native_sources(self):
        console.log("\n".join(OneFile.list_files(f'''{self.base_dir}/xcompiled-src/native''', True)))
    
    def get_test_cases(self):
        return [testCase.SyncTestCase("ListXCompiledNativeSources", lambda _: self.list_xcompiled_native_sources())]

File: PHP/src/OneLang/Test/TestCases/OneFileTests.php
<?php

namespace OneLang\Test\TestCases\OneFileTests;

use OneLang\File\OneFile;
use OneLang\Test\TestCase\ITestCollection;
use OneLang\Test\TestCase\SyncTestCase;
use OneLang\Test\TestCase\TestCase;

class OneFileTests implements ITestCollection {
    public $name = "OneFileTests";
    public $baseDir;
    
    function __construct($baseDir) {
        $this->baseDir = $baseDir;
    }
    
    function listXCompiledNativeSources() {
        \OneLang\Core\console::log(implode("\n", OneFile::listFiles($this->baseDir . "/xcompiled-src/native", true)));
    }
    
    function getTestCases() {
        return array(new SyncTestCase("ListXCompiledNativeSources", function ($_) { return $this->listXCompiledNativeSources(); }));
    }
}
